core/crypto/signed_hdr.c

```diff
@@ -60,6 +60,57 @@ static bool is_weak_hash_algo(uint32_t algo)
 	       algo == TEE_ALG_SHA224;
 }
 
+/*
+ * Checks for key sizes in bits, that are either depracted or put into legacy
+ * by national standards bodies. For deprecated key sizes return true. For
+ * legacy key sizes, return false, but print an informative message if
+ * CFG_WARN_INSECURE was set when compiling the TEE.
+ */
+static bool is_weak_key_size(uint32_t algo, size_t key_size_bits)
+{
+	switch (algo) {
+	case TEE_ALG_RSASSA_PKCS1_V1_5_MD5:
+	case TEE_ALG_RSASSA_PKCS1_V1_5_SHA1:
+	case TEE_ALG_RSASSA_PKCS1_V1_5_SHA224:
+	case TEE_ALG_RSASSA_PKCS1_V1_5_SHA256:
+	case TEE_ALG_RSASSA_PKCS1_V1_5_SHA384:
+	case TEE_ALG_RSASSA_PKCS1_V1_5_SHA512:
+	case TEE_ALG_RSASSA_PKCS1_V1_5_MD5SHA1:
+	case TEE_ALG_RSASSA_PKCS1_V1_5_SHA3_224:
+	case TEE_ALG_RSASSA_PKCS1_V1_5_SHA3_256:
+	case TEE_ALG_RSASSA_PKCS1_V1_5_SHA3_384:
+	case TEE_ALG_RSASSA_PKCS1_V1_5_SHA3_512:
+	case TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA1:
+	case TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA224:
+	case TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA256:
+	case TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA384:
+	case TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA512:
+	case TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA3_224:
+	case TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA3_256:
+	case TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA3_384:
+	case TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA3_512:
+	case TEE_ALG_RSAES_PKCS1_V1_5:
+	case TEE_ALG_RSAES_PKCS1_OAEP_MGF1_SHA1:
+	case TEE_ALG_RSAES_PKCS1_OAEP_MGF1_SHA224:
+	case TEE_ALG_RSAES_PKCS1_OAEP_MGF1_SHA256:
+	case TEE_ALG_RSAES_PKCS1_OAEP_MGF1_SHA384:
+	case TEE_ALG_RSAES_PKCS1_OAEP_MGF1_SHA512:
+	case TEE_ALG_RSAES_PKCS1_OAEP_MGF1_SHA3_224:
+	case TEE_ALG_RSAES_PKCS1_OAEP_MGF1_SHA3_256:
+	case TEE_ALG_RSAES_PKCS1_OAEP_MGF1_SHA3_384:
+	case TEE_ALG_RSAES_PKCS1_OAEP_MGF1_SHA3_512:
+	case TEE_ALG_RSA_NOPAD:
+		if (key_size_bits <= 2048)
+			return true;
+
+		if (IS_ENABLED(CFG_WARN_INSECURE) && key_size_bits < 3000)
+			IMSG("WARNING: Using legacy key size for RSA");
```

Comment:
This warning doesn't make sense. The problem is that we accept weak keys, not whether this particular key is weak. If we need to warn about this, it should be done `boot_init_primary_runtime()` when we complain about `CFG_INSECURE`.

Reply:
I removed it.